upgrade honnef.co/go/tools

otan opened the pull request:
Seems flaky for TestStaticCheck, which doesn't seem to pick up test
files referencing functions some of the time. Maybe an upgrade will help.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50617)
<!-- Reviewable:end -->

otan committed:
upgrade honnef.co/go/tools

Seems flaky for TestStaticCheck, which doesn't seem to pick up test
files referencing functions some of the time. Maybe an upgrade will help.

Release note: None
